src/test/java/app/daos/GenreDAOTest.java

```diff
@@ -1,68 +1,71 @@
-package org.example.daos;
+package app.daos;
 
 import jakarta.persistence.EntityManager;
 import jakarta.persistence.EntityManagerFactory;
 import jakarta.persistence.EntityNotFoundException;
-import org.example.config.HibernateConfig;
-import org.example.dtos.GenreDTO;
-import org.junit.jupiter.api.Assertions;
-import org.junit.jupiter.api.BeforeAll;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import app.config.HibernateConfig;
+import app.dtos.GenreDTO;
+import org.junit.jupiter.api.*;
 
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
 class GenreDAOTest {
-    private static EntityManagerFactory emf;
+    private static EntityManagerFactory emfTest;
     private static GenreDAO genreDAO;
 
-    private List<GenreDTO> genreDTOS;
+    private List<GenreDTO> genreDTOs;
 
     @BeforeAll
     static void beforeAll() {
-        emf = HibernateConfig.getEntityManagerFactoryForTest();
-        genreDAO = new GenreDAO(emf);
+        emfTest = HibernateConfig.getEntityManagerFactoryForTest();
+        genreDAO = new GenreDAO(emfTest);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        emfTest.close();
```

Comment:
Remove tearDown closing the entity manager factory. The factory is a singleton within HibernateConfig so closing it will prevent other test classes from running. The program terminates after tests anyway so it's probably not necessary to close it

Reply:
Agree that a teardown i not entirely necessary in our case since we make use of try-with-resources in setup. But closing it should not have an effect on other test classes since they do get created again in each test class and the idea was to make sure that, when working on same database, that there's isolation between each test class and we start from a clean slate. But it is an expensive operation between each class and should technically only be created and closed before and after the entire test suite.

Reply:
- See my comment on MovieDAOTest (above)